parlai/core/params.py

```diff
@@ -138,6 +138,19 @@ def add_parlai_args(self):
             '-bs', '--batchsize', default=1, type=int,
             help='batch size for minibatch training schemes')
         self.add_parlai_data_path(parlai)
+        # Find which task specified, and add its specific arguments.
+        args = sys.argv
+        task = None
+        for index, item in enumerate(args):
+            if item == '-t' or item == '--task':
+                task = args[index + 1]
+        if task:
+            agent = get_agent_module(task)
+            if hasattr(agent, 'add_cmdline_args'):
+                agent.add_cmdline_args(self)
+            if hasattr(agent, 'dictionary_class'):
```

Comment:
The task agent is not a "model" (whose cmdline args are added in `add_model_args`, so unfortunately its `add_cmdline_args` is never called for `Teacher` agents unless we add this.